src/main/java/com/vesanieminen/froniusvisualizer/views/NordpoolspotView.java

```diff
@@ -449,13 +454,26 @@ private DataSeries createEstimateDataSeries(List<FingridLiteResponse> dataSource
         return dataSeries;
     }
 
-    private DataSeries createTemperatureDataSeries(List<SpotHintaResponse> dataSource, String title) {
+    private DataSeries createTemperatureDataSeries(List<SpotHintaResponse> spotHintadataSource, FmiObservationResponse fmiObservations, String title) {
         final var dataSeries = new DataSeries(title);
-        for (SpotHintaResponse response : dataSource) {
-            final var dataSeriesItem = new DataSeriesItem();
-            dataSeriesItem.setX(response.TimeStamp.toInstant().plus(Duration.ofHours(2)));
-            dataSeriesItem.setY(response.Temperature);
-            dataSeries.add(dataSeriesItem);
+
+        if (spotHintadataSource != null) {
+            for (SpotHintaResponse response : spotHintadataSource) {
+                final var dataSeriesItem = new DataSeriesItem();
+                dataSeriesItem.setX(response.TimeStamp.toInstant().plus(Duration.ofHours(2)));
```

Comment:
I don't know if this addition of two hours in meant to be a timezone conversion? I think the observations might have the same offset issue.

Reply:
I think that's a correct guess, as currently in liukuri.fi the forecasts start from now-1 hours.

Reply:
Thanks for the find. This is something I wrote very early on in order to get things working. I'll have to rewrite that timezone conversion in a better way.